### PR TITLE
Add services for calendar

### LINKED ISSUE #444: Add services for Calendar page to AdminPanel
AP Client should be able to obtain filtered scheduled events (POST /api/schedules/events)

### PR DESCRIPTION
**Pull request by gingerbread-monster**
With this commits I'm adding implementation to services that provide required data for calendar page (that is to be added in https://github.com/ita-social-projects/WhatBackend/issues/445).

Web API project:
* Added new endpoint to API: `api/events/event-occurrences` that returns all event occurrences

Admin panel:
* Added and implemented `IScheduleService`
* Added and implemented `ICalendarService`
* Added corresponding ViewModels & mappings for them

### CODE REVIEW

File: CharlieBackend.AdminPanel/Controllers/CalendarController.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            _calendarService = calendarService;
+        }
+
+        public async Task<IActionResult> Index(
+            ScheduledEventFilterRequestDTO scheduledEventFilter)
+        {
+            var data = await _calendarService.GetCalendarDataAsync(scheduledEventFilter);
+
+            return Ok(data);
+        }
```
Comment (gingerbread-monster) [inline CharlieBackend.AdminPanel/Controllers/CalendarController.cs:25]: Temporary implementation that is to be changed in https://github.com/ita-social-projects/WhatBackend/issues/445.
Tested:
tested

**Review by Serwlashev — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by zelya2021 — APPROVED** [review]

**Review by TetianaDomanska-dev — APPROVED** [review]

**Review by Serhii-Dudar — APPROVED** [review]

File: CharlieBackend.AdminPanel/Services/CalendarService.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+            _scheduledEventFilter = scheduledEventFilter;
+        }
+
+        /// <summary>
+        /// Makes HTTP calls to Web API and obtains required data.
+        /// </summary>
+        private async Task GetDataFromApiAsync()
+        {
+            _activeCourseDtos = await _apiUtil.GetAsync<IList<CourseDto>>("api/courses/isActive");
```
Comment (mahotskuy) [inline CharlieBackend.AdminPanel/Services/CalendarService.cs]: Move getting data and mapping to a methods. Then use Task.WhenAll to make request simultaneously and then create a returned ViewModel. Please remove  _activeCourseDtos  and other fields.

Comment (gingerbread-monster) [reply CharlieBackend.AdminPanel/Services/CalendarService.cs]: Fixed in https://github.com/ita-social-projects/WhatBackend/pull/460/commits/9fe3bf8d9c8947ee7f956338d0d565595fe6e6b7
Now all data obtaining tasks run not one by one but as parallel async tasks.

File: CharlieBackend.AdminPanel/Services/CalendarService.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        /// </summary>
+        /// <param name="scheduledEventFilter">DTO with an optional set of 
+        /// filters for scheduled events.</param>
+        /// <returns>>model with data required for viewing a calendar.</returns>
+        public async Task<CalendarViewModel> GetCalendarDataAsync(
+            ScheduledEventFilterRequestDTO scheduledEventFilter)
+        {
+            ApplyDefaultDateEventFilter(scheduledEventFilter);
+
+            var getCoursesTask = Task.Run(() => GetActiveCourseViewModelsAsync());
```
Comment (mahotskuy) [inline CharlieBackend.AdminPanel/Services/CalendarService.cs:56]: GetActiveCourseViewModelsAsync is already a Task, so you can skip Task.Run()

Comment (gingerbread-monster) [reply CharlieBackend.AdminPanel/Services/CalendarService.cs:56]: I'm sorry I merged this pull request too soon.
Removing redundant `Task.Run()` calls in https://github.com/ita-social-projects/WhatBackend/pull/483

**Pull request merged**